Metalama.Framework.DesignTime/Utilities/DesignTimeExceptionHandler.cs

```diff
@@ -19,15 +20,17 @@ internal static bool MustHandle( Exception e )
                 _ => true
             };
 
-        public static void ReportException( Exception e, ILogger? logger = null )
+        public static void ReportException( Exception e, IExceptionReporter? exceptionReporter, ILogger? logger = null )
```

Comment:
This class is a smell. If it depends on a service, it must be made a service itself.

So make this class non-static, add it as a service, with IExceptionReporter as an optional dependency.